chore: adjust rules for using tabs instead of spaces

tomislav-s opened the pull request:
* removes preference from `.editorconfig`
* replaces `react/jsx-indent`, as well as `react/jsx-indent-props` with tabs instead of spaces

heldinz reviewed (changes requested):
Mostly looking good! Could we please stick with the `'error'` format rather than `2` to set the error level for the rule though? I find the numbers can be confusing, especially when coupled with things like `tab` and `space`.

Btw, I also noticed (while checking out the `base-next` changes locally) that we need to update our eslint config to not require React imports anymore: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md?ts=8#when-not-to-use-it

Would you mind updating this as well, before we do a release? 🙏🏻

tomislav-s commented:
> Mostly looking good! Could we please stick with the `'error'` format rather than `2` to set the error level for the rule though? I find the numbers can be confusing, especially when coupled with things like `tab` and `space`.
> 
> Btw, I also noticed (while checking out the `base-next` changes locally) that we need to update our eslint config to not require React imports anymore: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md?ts=8#when-not-to-use-it
> 
> Would you mind updating this as well, before we do a release? 🙏🏻

Thanks for catching this! Replaced `2` with `error` and added support for React 17 JSX transform

heldinz reviewed (approved):
Nice, thanks! 👏🏻
